pkg/controller/storagecluster/storagecluster.go

```diff
@@ -247,8 +247,9 @@ func (c *Controller) Reconcile(ctx context.Context, request reconcile.Request) (
 }
 
 func (c *Controller) waitingForMigrationApproval(cluster *corev1.StorageCluster) bool {
-	approved, err := strconv.ParseBool(cluster.Annotations[constants.AnnotationMigrationApproved])
-	return err == nil && !approved
+	_, migrationLabelPresent := cluster.Annotations[constants.AnnotationMigrationApproved]
+	return (migrationLabelPresent && cluster.Status.Phase == "") ||
```

Comment:
isn't cluster status waitingForApproval derived from the annotation, why do we need this change?

Reply:
There was a race condition where the operator reconcile loop was starting to deploy px pods before the migration handler could add the node labels to prevent the operator from starting px pods right away. Although the pods immediately get terminated but there are a lot of pods created and terminated for few seconds.

This makes sure that the operator reconcile loop waits for the migration to start and only then it is starts allowing the StorageCluster waiting to be migrated.